sql/wsrep_mysqld.cc

```diff
@@ -835,10 +835,18 @@ static std::string wsrep_server_incoming_address() {
        !strcmp(wsrep_node_incoming_address, WSREP_NODE_INCOMING_AUTO))) {
     bool is_ipv6 = false;
     unsigned int my_bind_ip = INADDR_ANY;  // default if not set
+    char *single_addr = nullptr;
 
     if (my_bind_addr_str && strlen(my_bind_addr_str) &&
         strcmp(my_bind_addr_str, "*") != 0) {
-      my_bind_ip = wsrep_check_ip(my_bind_addr_str, &is_ipv6);
+      size_t single_addr_buf_size = strlen(my_bind_addr_str) + 1;
+      single_addr = (char *)malloc(single_addr_buf_size);
```

Comment:
Here and for all occurences in this patch:

We need to instrument every dynamic memory allocation so that it can be observed through perfschema.
Please use `my_malloc` instead of `malloc` and `my_free` instead of `free`
```suggestion
      single_addr = (char *) my_malloc(key_memory_wsrep, single_addr_buf_size, MYF(0));
```

Reply:
I have hanged it to my_malloc, but as we discussed on Slack I don't see a big advantage here.
my_malloc add instrumentation can be useful in non-trival scenarios (we allocate memory here, free it somewhere else, etc). Here we do malloc/do_quick_simple_task/free with no possibility to malloc but not free. I'm not sure if there is any advantage in instrumenting such allocations. It is comparable to having  std::string object on the stack, which memory is not instrumented as well. 

Additionally in my opinion it should not be the rule of thumb that we should always use my_malloc over malloc.
Internally my_malloc/my_free  use atomics, so if we use them in places executed in parallel by several threads we will pay the performance penalty. In such cases, it is better to use malloc/free. my_malloc/my_free should be used for the memory we want to track (e.g. not trivial allocation/free patterns), not everywhere.